Modules/mmapmodule.c

```diff
@@ -682,6 +682,86 @@ mmap__exit__method(PyObject *self, PyObject *args)
     return _PyObject_CallMethodId(self, &PyId_close, NULL);
 }
 
+static PyObject *
+mmap__repr__method(PyObject *self)
+{
+    PyObject *repr;
+    mmap_object *m_obj;
+    m_obj = (mmap_object *) self;
+
+    const char *reprfmt;
+    const char *access_str;
+
+    switch (m_obj->access) {
+        case ACCESS_DEFAULT:
+            access_str = "ACCESS_DEFAULT";
+            break;
+
+        case ACCESS_READ:
+            access_str = "ACCESS_READ";
+            break;
+
+        case ACCESS_WRITE:
+            access_str = "ACCESS_WRITE";
+            break;
+
+        case ACCESS_COPY:
+            access_str = "ACCESS_COPY";
+            break;
+
+        default:
+            PyErr_SetString(PyExc_IOError, "Unknown access mode for mmap object.");
+            return NULL;
+    }
+
+    Py_ssize_t size = m_obj->size;
+    const char *tp_name = self->ob_type->tp_name;
+
+
+#ifdef MS_WINDOWS
+    int fileno = m_obj->file_handle;
+    if (m_obj->map_handle == NULL)
+#elif defined(UNIX)
+    int fileno = m_obj->fd;
+    if (m_obj->data == NULL)
+#endif
+    {
+        reprfmt = "<%s is_closed=True fileno=%d access=%s>";
+        repr = PyUnicode_FromFormat(reprfmt, tp_name, fileno, access_str);
+    }
+    else {
+        const char *data = m_obj->data;
+        Py_ssize_t pos = m_obj->pos;
+
+        PyObject *length = PyLong_FromSize_t(size);
+        PyObject *offset = PyLong_FromSize_t(pos);
+
+        if (size < 64) {
+            reprfmt = "<%s is_closed=False fileno=%d access=%s size=%R "
+                      "offset=%R entire_contents=%R>";
+
+            PyObject *entire_contents;
+            entire_contents = PyBytes_FromStringAndSize(data, size);
+
+            repr = PyUnicode_FromFormat(reprfmt, tp_name, fileno, access_str,
+                                        length, offset, entire_contents);
+        }
+        else {
+            reprfmt = "<%s is_closed=False fileno=%d access=%s size=%R "
+                      "offset=%R entire_contents=%R ... %R>";
+
+            PyObject *slice1 = PyBytes_FromStringAndSize(data, 32);
+            PyObject *slice2 = PyBytes_FromStringAndSize(data + size - 32,
+                                                         32);
+
+            repr = PyUnicode_FromFormat(reprfmt, tp_name, fileno, access_str,
+                                        length, offset, slice1, slice2);
+        }
```

Comment:
I don't think this part is necessary. IMHO we should not show the content in repr.

Reply:
The original proposer of this said he felt like to have these ones:
 - Length
 - Whether it's a file or anonymous memory
 - The entire contents
 - The clipped contents
 - Whether it's opened or closed
 - The access type

I think having entire contents is infeasible, but others seem to be reasonable.